packages/headless/src/controllers/product-listing/headless-product-listing.ts

```diff
@@ -22,19 +24,74 @@ const optionsSchema = new Schema({
   }),
 });
 
-export type ProductListingOptions = SchemaValues<typeof optionsSchema>;
+export interface ProductListingOptions {
+  /**
+   * The initial url used to retrieve the product listing.
+   */
+  url: string;
+
+  /**
+   * The initial additional fields to retrieve with the product listing.
+   */
+  additionalFields: string[];
+}
 
 export interface ProductListingProps {
+  /**
+   * The initial options that should be applied to this `ProductListing` controller.
+   */
   options?: ProductListingOptions;
 }
 
-export type ProductListingController = ReturnType<typeof buildProductListing>;
+/**
+ * The `ProductListing` headless controller allows the end user to configure and retrieve product listing data.
+ */
+export interface ProductListingController extends Controller {
```

Comment:
Typically, controller interfaces do not end with `Controller`. Would it be possible to avoid this? Could just `ProductList` (analogue of `ResultList` for search pages) work just as well? Other ideas:

- `ProductListingCatalogue`
- `ProductListingResultList`
- `ProductListing` (though this one sounds strange imo)

I feel we might have talked about this before, I vaguely remember a `ProductListingList` interface name hmm.

Reply:
It really is `ProductListing`, not `Product List`, it's a specific commerce lingo :smile: